Apps/Database/src/Delegates/DBCommunicationDelegate.cs

```diff
@@ -181,5 +216,51 @@ public DBResult<Communication> Delete(Communication communication, bool commit =
             this.logger.LogDebug($"Finished deleting Communication in DB");
             return result;
         }
+
+        private List<Communication> GetActiveBannerCommunications()
+        {
+            List<Communication> communication = this.dbContext.Communication
+                 .OrderByDescending(c => c.CreatedDateTime)
+                 .Where(c => c.CommunicationTypeCode == CommunicationType.Banner
+                     && c.CommunicationStatusCode == CommunicationStatus.New && DateTime.UtcNow <= c.ExpiryDateTime)
+                 .ToList();
+
+            return communication;
+        }
+
+        private bool IsCommunicationValidCommit(Communication communication)
+        {
+            List<Communication> activeBannerCommunication = this.GetActiveBannerCommunications();
+
+            if (communication.CommunicationStatusCode == CommunicationStatus.New)
+            {
+                if (activeBannerCommunication != null && activeBannerCommunication.Count == 0)
+                {
+                    return true;
+                }
+                else
+                {
+                    var validCommunications = activeBannerCommunication
+                            .OrderByDescending(c => c.EffectiveDateTime)
+                            .Where(c => (c.ExpiryDateTime < communication.ExpiryDateTime &&
+                                        communication.EffectiveDateTime > c.ExpiryDateTime) ||
+                                        (communication.EffectiveDateTime < c.EffectiveDateTime && communication.ExpiryDateTime < c.EffectiveDateTime))
+                            .ToList();
```

Comment:
Is it necessary to bring back a list of Communication objects?  It looks like you want to check if Communications exist.  Is it possible to use Any<> or Count<> instead of ToList()?  Any<> would return a bool and Count would return an int.

Reply:
@BrianMaki , I am waiting for UI feedback, if we need to show the information like recent effective date or expiry date. I

Reply:
@pagustinqt20 I'm not sure about the dates if @ninosamson would require that information but I did rewrote the error message to reflect the bug "Banner post could not be added because there is an existing banner post."
Also we can just go with one version and if there are any changes we can do it in the next iteration.

Reply:
@mk08sh  Thanks. Will refactor the code based on static message.